eva_submission/submission_qc_checks.py

```diff
@@ -0,0 +1,316 @@
+        failed_files = {}
+        for file in vcf_files:
+            accessioning_log_files = glob.glob(f"{self.path_to_data_dir}/00_logs/accessioning.*{file}*.log")
+            if accessioning_log_files:
+                # check if accessioning job completed successfullyy
+                if not self.check_if_job_completed_successfully(accessioning_log_files[0]):
+                    failed_job = self.check_for_errors_in_case_of_job_failure(accessioning_log_files[0])
+                    failed_files[
+                        file] = f"failed_job - {failed_job}" if failed_job else f"failed_job name could not be retrieved"
+            else:
+                failed_files[file] = f"error : No accessioning file found for {file}"
+
+        self._accessioning_job_check_result = "PASS" if not failed_files else "FAIL"
+        report = f"""
+                pass: {self._accessioning_job_check_result}"""
+        if failed_files:
+            report += f"""
+                failed_files:"""
+            for file, value in failed_files.items():
+                report += f"""
+                    {file} - {value}"""
+
+        return report
+
+    def check_if_variant_load_completed_successfully(self, vcf_files):
+        failed_files = {}
+        for file in vcf_files:
+            pipeline_log_files = glob.glob(f"{self.path_to_data_dir}/00_logs/pipeline.*{file}*.log")
+            if pipeline_log_files:
+                # check if variant load job completed successfully
+                if not self.check_if_job_completed_successfully(pipeline_log_files[0]):
+                    failed_job = self.check_for_errors_in_case_of_job_failure(pipeline_log_files[0])
+                    failed_files[
+                        file] = f"failed_job - {failed_job}" if failed_job else f"failed_job name could not be retrieved"
+            else:
+                failed_files[file] = f"error : No pipeline file found for {file}"
+
+        self._variant_load_job_check_result = "PASS" if not failed_files else "FAIL"
+        report = f"""
+                pass: {self._variant_load_job_check_result}"""
+        if failed_files:
+            report += f"""
+                failed_files:"""
+            for file, value in failed_files.items():
+                report += f"""
+                    {file} - {value}"""
+
+        return report
+
+    def check_if_variants_were_skipped_while_accessioning(self, vcf_files):
+        failed_files = {}
+        for file in vcf_files:
+            accessioning_log_files = glob.glob(f"{self.path_to_data_dir}/00_logs/accessioning.*{file}*.log")
+            if accessioning_log_files:
+                # check if any variants were skippped while accessioning
+                variants_skipped = self.check_if_variants_were_skipped(accessioning_log_files[0])
+                if variants_skipped:
+                    failed_files[file] = f"{variants_skipped} variants skipped"
```

Comment:
I think there are legitimate reason why the variants might be skipped. Are you sure we want to fail in this case ?

Reply:
can we put something like **PASS with Warning**  or just **Warning** or **manual check required**?
I was not sure about this since we have only PASS or FAIL as status. If no variants are skipped, it is definitely a PASS.
If variants are skipped we don't know if they were legitimate or not unless we check, if we don't want FAIL in that condition by default, it will never have the FAIL result.